File: src/main/java/com/procurement/submission/model/dto/request/AwardStatusDetails.java
package com.procurement.submission.model.dto.request;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public enum AwardStatusDetails {
    PENDING("pending"),
    ACTIVE("active"),
    UNSUCCESSFUL("unsuccessful"),
    EMPTY("empty");

    private static final Map<String, AwardStatusDetails> CONSTANTS = new HashMap<>();
    private final String value;

    static {
        for (final AwardStatusDetails c : values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    AwardStatusDetails(final String value) {
        this.value = value;
    }

    @JsonCreator
    public static AwardStatusDetails fromValue(final String value) {
        final AwardStatusDetails constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new IllegalArgumentException(
                    "Unknown enum type " + value + ", Allowed values are " + Arrays.toString(values()));
        }
        return constant;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

}

File: src/main/kotlin/com/procurement/submission/domain/model/enums/AwardStatusDetails.kt
package com.procurement.submission.domain.model.enums

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class AwardStatusDetails(@JsonValue override val key: String) : EnumElementProvider.Key {
    ACTIVE("active"),
    AWAITING("awaiting"),
    BASED_ON_HUMAN_DECISION("basedOnHumanDecision"),
    CONSIDERATION("consideration"),
    EMPTY("empty"),
    LOT_CANCELLED("lotCancelled"),
    NO_OFFERS_RECEIVED("noOffersReceived"),
    PENDING("pending"),
    UNSUCCESSFUL("unsuccessful");

    override fun toString(): String = key

    companion object : EnumElementProvider<AwardStatusDetails>(info = info()) {

        @JvmStatic
        @JsonCreator
        fun creator(name: String) = AwardStatusDetails.orThrow(name)
    }
}